src/api/EscargotPublic.h

```diff
@@ -679,6 +679,44 @@ class ESCARGOT_EXPORT VMInstanceRef {
     Evaluator::EvaluatorResult executePendingJob();
 };
 
+class ESCARGOT_EXPORT DebuggerOperationsRef {
+public:
+    class WeakCodeRef;
+
+    struct BreakpointLocation {
+        BreakpointLocation(uint32_t line, uint32_t offset)
+            : line(line)
+            , offset(offset)
+        {
+        }
+
+        uint32_t line;
+        uint32_t offset;
+    };
+
+    typedef std::vector<BreakpointLocation> BreakpointLocationVector;
+
+    struct BreakpointLocationsInfo {
+        BreakpointLocationsInfo(WeakCodeRef* weakCodeRef)
+            : weakCodeRef(weakCodeRef)
+        {
+        }
+
+        // The codeRef is a unique id which is not garbage collected
+        // to avoid keeping script / function code in the memory.
+        WeakCodeRef* weakCodeRef;
```

Comment:
This member should not be tracked by GC. How can I do this?

Reply:
Exactly. The point is, as long as any byte code exists, the debugger should know about it. But it should not have a hard reference to it, because it would prevent freeing it forever. Instead, `codeReleased()` callback tells when a pointer becomes invalid, and should not be used again.

Reply:
(To make things clear) `codeReleased` should be triggered when bytecode is no longer valid (garbage collected), right?.
But we cannot know exactly the validity of any bytecode (only GC knows it during the garbage collection).
In that case, `codeReleased` should be invoked from GC. Is this what you want?

Reply:
Yes. It already does this.

Reply:
Ok, then `codeReleased` simply needs to set `WeakCodeRef` as null pointer or something elase to denote that this pointer is invalid now.
Since `BreakpointLocationsInfo` and 'WeakCodeRef` are not allocated by GC, GC will not trace them, so it would not interfere the GC operation as a result.